File: C++/0060.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        string result = "";
        string numbers = "123456789";

        vector<int> factorial(n, 1);
        for (int i = 2; i < n; ++i) {
            factorial[i] = i * factorial[i - 1];
        }

        --k; // for index
        for (int i = n; i >= 1; --i) {
            int j = k / factorial[i - 1];
            k %= factorial[i - 1];
            result += numbers[j];
            numbers.erase(j, 1);
        }

        return result;
    }
};

File: Python/0060.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        result = ''
        numbers = [str(i) for i in range(1, 11)]
        
        factorial = [1] * n
        for i in range(2, len(factorial)):
            factorial[i] = i * factorial[i - 1]
        
        # for index
        k -= 1
        for i in range(n, 0, -1):
            j = k // factorial[i - 1]
            k %= factorial[i - 1]
            result += numbers[j]
            del numbers[j]

        return result
